Sources/Core/Sources/Render/RenderCoordinator.swift

```diff
@@ -112,17 +125,21 @@ public final class RenderCoordinator: NSObject, MTKViewDelegate {
     let time = CFAbsoluteTimeGetCurrent()
     let frameTime = time - previousFrameStartTime
     previousFrameStartTime = time
-
-    profiler.push(.waitForRenderPassDescriptor)
-    // Get current render pass descriptor
-    guard let renderPassDescriptor = view.currentRenderPassDescriptor else {
-      log.error("Failed to get the current render pass descriptor")
-      client.eventBus.dispatch(ErrorEvent(
-        error: RenderError.failedToGetCurrentRenderPassDescriptor,
-        message: "RenderCoordinator failed to get the current render pass descriptor"
-      ))
+    
+    profiler.push(.updateRenderTarget)
+    do {
+      try screenRenderer.updateRenderTarget(for: view)
+    } catch {
+      log.error("Failed to update render target: \(error)")
+      client.eventBus.dispatch(ErrorEvent(error: error, message: "Failed to update render target"))
       return
     }
+    
+    profiler.pop()
+    
+    profiler.push(.waitForOffscreenRenderPassDescriptor)
```

Comment:
Would this measurement always be basically negligible? If so there's probably no need to add it to the profiler

Reply:
At the moment - no. It was mostly done with buffering in mind. If we were to introduce such concept, there could be a non-negligible time needed to obtain free descriptor for current frame. I'll go ahead and remove it for now.